WIP Document realm export (and sequestered organization) feature (#9781)
Scille/parsec-cloud: docs/realm_export.md

Sequester service certificate Service 1 (id: ef9adae7ee9f44cc9f974fdcaaff8839, timestamp: 2025-02-23T21:19:35.484948Z) exported in ./sequester_service_certificate-ef9adae7ee9f44cc9f974fdcaaff8839-2025-02-23T21:19:35.484948Z.pem
Use parsec sequester create_service command to add it to an organization
# 3. Create the sequester service on the server using the certificate
$ parsec sequester create_service --db=$POSTGRESQL_URL --organization=$ORGANIZATION_ID --service-certificate=./sequester_service_certificate-ef9adae7ee9f44cc9f974fdcaaff8839-2025-02-23T21:19:35.484948Z.pem
Service created
```

> Note:
>
> Once added, the sequester service cannot decrypt any data since no realm key has been
> encrypted with its public key yet.
>
> However the sequester service creation is detected by all Parsec client with a OWNER
> role in the workspace and they automatically trigger a realm key rotation in order
> to ensure the sequester service gets access to the workspace data.
>
> The same behavior is also triggered whenever the sequester service is revoked, this

> > time in order to ensure any new data added to teh workspace won't be accessible to

Reviewer: ```suggestion
> time in order to ensure any new data added to the workspace won't be accessible to
```

```

## 3 - Do a realm export

The realm export is done with the server CLI:

```shell
parsec export_realm --db=$POSTGRESQL_URL --blockstore=$S3_URL --organization=$ORGANIZATION_ID --realm=$REALM_ID
```

This creates a file `parsec-export-$ORG-realm-$REALM-$TIMESTAMP.sqlite`.

> Notes:
>
> - The realm export can be limited up to a specific point in time using the
>   `--snapshot-timestamp` option.
> - A stopped export operation (using ^C) can be restarted by passing the output
>   file as argument along with the time of the initial export with `--snapshot-timestamp`.

## 4 - Read the realm export

The realm export file should then be transfered to a secured machine containing the
decryption keys.

The decryption keys can be of two types:

- Sequestered service key in case of a sequestered organization.
- Device key otherwise (i.e. decrypt the realm export from a machine normally
  used to run the Parsec client).

It is recommended to use a dedicated machine for this operation with no access to the
internet.

And from there the client CLI should be used to decrypt the workspace
from realm export file and mount it as a local directory.

> Also note a `--timestamp=<TIMESTAMP_AS_RFC3339>` option can be privided to specify

Reviewer: ```suggestion
Also note a `--timestamp=<TIMESTAMP_AS_RFC3339>` option can be provided to specify
```

Author: :+1:

Realm ID: 9275cdee-7a19-42cf-ae35-7ebb945f71a6
Export temporal bounds: 2025-02-23T20:29:07.603857Z to 2025-02-23T21:47:59.873115Z
polling for the start...
mountpoint is ready !
Mounted at "./parsec-export-TestOrg-realm-9275cdee7a1942cfae357ebb945f71a6-20250223T214759Z"
```

> Note:
>
> The sequester service key is expected to be provided as a PEM file containing
> the key decrypted (i.e. file must start with `-----BEGIN PRIVATE KEY-----`).

### Decrypt with a device

```shell
$ parsec-cli mount-realm-export --decryptor device:$DEVICE_ID ./parsec-export-TestOrg-realm-9275cdee7a1942cfae357ebb945f71a6-20250223T214759Z.sqlite
Enter password for the device:
Organization: TestOrg
Realm ID: 9275cdee-7a19-42cf-ae35-7ebb945f71a6
Export temporal bounds: 2025-02-23T20:29:07.603857Z to 2025-02-23T21:47:59.873115Z
polling for the start...
mountpoint is ready !
Mounted at "./parsec-export-TestOrg-realm-9275cdee7a1942cfae357ebb945f71a6-20250223T214759Z"
```

> Note:
>
> Multiple keys can be provided (by passing multiple `--decryptor` options). This is not
> useful under normal conditions since any sequester service / user with access to the
> workspace is able to decrypt all it data.

> > However it may come handy in case of a partially corrupter workspace (e.g. in case a
> > mallicious/buggy actor has provided corrupted keys during a realm key rotation).

Reviewer: ```suggestion
> However it may come handy in case of a partially corrupted workspace (e.g. in case a
> malicious/buggy actor has provided corrupted keys during a realm key rotation).
```

Author: :+1:
